src/kaleidoscope/plugin/LEDModeInterface.h

```diff
@@ -16,13 +16,32 @@
 
 #pragma once
 
+#include <Arduino.h>  // for __FlashStringHelper, F
+
+#include "kaleidoscope/event_handler_result.h"  // for EventHandlerResult
+#include "kaleidoscope/event_handlers.h"        // for LedModeCallback
+
 namespace kaleidoscope {
 namespace plugin {
 
 class LEDModeInterface {
  public:
+  LEDModeInterface()
+    : ledModeName{0} {}
+  explicit LEDModeInterface(const __FlashStringHelper *_ledModeName)
+    : ledModeName{_ledModeName} {}
   void activate();
 
+  EventHandlerResult onLedEffectQuery(LedModeCallback callback) {
+    if (ledModeName == 0) {
+      // If no name was defined, return a default string
+      callback(F("[unnamed led mode]"));
+    } else {
+      callback(ledModeName);
+    }
```

Comment:
I'm wondering if there might an argument for moving this logic into the callback function. I think that this is the more obvious place to do a default-if-unset test, but I'm curious what you think.

Reply:
That makes sense to me.